gitpod-io/gitpod
# [installer] Add a second `slow-server` deployment behind `experimental.webapp.slowDatabase` flag

## Description

Add a new deployment, `slow-server`, identical to `server` except that the database connection goes through Toxiproxy.

The new deployment is only rendered if the `experimental.webapp.slowDatabase` flag is set in the installer config.

The new component is a copy of the existing `server` component with only minimal changes.

## Related Issue(s)

Part of #9198.

## How to test

* In the browser console for the preview environment run:

```js
await window._gp.gitpodService.server.adminGetBlockedRepositories({offset: 0, limit: 10, orderBy: 'id', orderDir: "asc"})
```

Note that the request has a very low latency.

* Edit the `proxy` deployment to run a version that has been configured to redirect all server traffic to the new `slow-server` deployment:

```
kubectl edit deploy proxy
<change the tag for the proxy container from `commit-be66c7b26c19a659278e7f86b46593623be57992` to `af-caddy-use-slow-server.0`> 
```
Unfortunately this step has to be done in code because `proxy` [hard-codes its config](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/Dockerfile#L35). The branch this uses is [here](https://github.com/gitpod-io/gitpod/compare/main...af/caddy-use-slow-server). 

* In the browser console for the preview environment run:

```js
await window._gp.gitpodService.server.adminGetBlockedRepositories({offset: 0, limit: 10, orderBy: 'id', orderDir: "asc"})
```
 
Observe that the request now has an additional ~1s of latency.

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
